rmw_fastrtps_shared_cpp/src/writer_info.hpp

```diff
@@ -95,19 +98,13 @@ class WriterInfo : public eprosima::fastrtps::rtps::ReaderListener
     mapmutex.unlock();
 
     if (trigger) {
-      rmw_ret_t ret = rmw_trigger_guard_condition(graph_guard_condition_);
-      if (ret != RMW_RET_OK) {
-        RCUTILS_LOG_ERROR_NAMED(
-          "rmw_fastrtps_cpp",
-          "failed to trigger graph guard condition: %s",
-          rmw_get_error_string_safe())
-      }
+      graph_guard_condition_->trigger();
```

Comment:
Where does the new code preserve the error message?

Reply:
See my answer above for reader_info.hpp